recommender/engines/nlp_embedders/handlers.py

```diff
@@ -0,0 +1,153 @@
+# pylint: disable=missing-class-docstring, missing-function-docstring, no-self-use
+
+"""
+Handlers used by object2vec - related functions. Used for embedding various
+kind of object fields.
+"""
+
+from typing import Any, List, Union, Callable
+
+import torch
+from cfgv import Optional
+from torch import Tensor
+from torch.nn.utils.rnn import pad_sequence
+
+from .text2vec import Text2Vec
+
+
+class Handler:
+    """
+    Used for embedding various kind of object fields in the
+    object2vec - related functions
+    """
+
+    def __init__(self, text2vec: Optional[Text2Vec] = None):
+        self.text2vec = text2vec or Text2Vec()
+
+    def __call__(self, obj: Any) -> None:
+        """
+        Can be subsequently called on similar objects. It builds so called
+        accumulator(s) that can be reduced into tensor(s) with the
+        `get_results` function.
+
+        Args:
+            obj: Iterable of objects
+        """
+
+    def get_results(self) -> List[Tensor]:
+        """
+        Based on accumulator(s) build with `__call__` function it reduce
+        it/them into list of tensors. Then it empties accumulator.
+        """
+
+    @property
+    def embedding_dim(self):
+        return self.text2vec.embedding_dim
+
+
+class StringHandler(Handler):
+    def __init__(self, field_name, text2vec=None):
+        super().__init__(text2vec)
+        self.field_name = field_name
+        self.accumulator = []
+
+    def __call__(self, obj):
+        self.accumulator.append(self.text2vec(getattr(obj, self.field_name)))
+
+    def get_results(self) -> List[Tensor]:
+        result = [torch.stack(self.accumulator)]
+        self.accumulator = []
+        return result
+
+
+class ListOfStringsHandler(Handler):
+    def __init__(self, field_name, text2vec=None):
+        super().__init__(text2vec)
+        self.field_name = field_name
+        self.accumulator = []
+
+    def __call__(self, obj):
+        self.accumulator.append(self.text2vec(", ".join(getattr(obj, self.field_name))))
+
+    def get_results(self) -> List[Tensor]:
+        result = [torch.stack(self.accumulator)]
+        self.accumulator = []
+        return result
+
+
+class ObjectHandler(Handler):
```

Comment:
No and for now it should be done in this way because 1) it is possible for now (no information loss) 2) it is very beneficial because it makes possible online learning from unseen examples